src/main/java/com/statful/client/framework/springboot/registry/StatfulMetricRegistry.java

```diff
@@ -0,0 +1,190 @@
+
+    private StatfulMetricsProperties properties;
+
+    public StatfulMetricRegistry(StepRegistryConfig config, StatfulClient statfulClient, Clock clock, StatfulMetricsProperties properties) {
+        super(config, clock);
+
+        this.statfulClient = statfulClient;
+        this.properties = properties;
+
+        applyCustomFilter();
+
+        start();
+    }
+
+    /**
+     * Applies two filters, one that looks into the property "acceptedMetrics" and another that looks into the "alias"
+     * and "tags" properties in {@link StatfulMetricsProperties}.
+     * <p>
+     * The first filter matches the prefixes defined in the list against the metrics. If the list is empty all metrics are
+     * accepted. If an alias has been applied to the metric beforehand then the filter will also search for the
+     * corresponding key in the "alias" property and accept the metric if found.
+     * <p>
+     * The second filter adds the tags to the metrics that match the keys in "tags" and replaces the name for the metrics
+     * that match the keys in "alias"
+     */
+    private void applyCustomFilter() {
+        this.config()
+                .meterFilter(
+                        MeterFilter.denyUnless(id ->
+                                properties.getAcceptedMetrics().isEmpty() || properties.getAcceptedMetrics().stream()
+                                        .anyMatch(metricName -> id.getName().startsWith(metricName) || isNotEmpty(getAliasForMetric(id)))))
+
+                .meterFilter(new MeterFilter() {
+
+                    @Override
+                    public Meter.Id map(Meter.Id id) {
+                        Meter.Id newId = id;
+
+                        for (Map.Entry<String, String> entry : properties.getTags().entrySet()) {
+                            if (id.getName().startsWith(entry.getKey())) {
+                                List<Tag> tags = toListOfTags(entry.getValue());
+
+                                newId = newId.withTags(tags);
+                            }
+                        }
+
+                        String metricAlias = getAliasForMetric(newId);
+
+                        if (isNotEmpty(metricAlias)) {
+                            newId = newId.withName(metricAlias);
+                        }
+
+                        return newId;
+                    }
+                });
+
+    }
+
+    private String getAliasForMetric(Meter.Id id) {
+        for (Map.Entry<String, String> entry : properties.getAlias().entrySet()) {
+            if (id.getName().startsWith(entry.getKey())) {
+                return entry.getValue();
+            }
+        }
+        return Strings.EMPTY;
+    }
+
+    private List<Tag> toListOfTags(String tags) {
+        String[] tagsAsList = tags.split(";");
+
+        return Arrays.stream(tagsAsList)
+                .map(tag -> {
+                    String[] splitTag = tag.split("=");
+
+                    return new Tag() {
```

Comment:
Why are you creating an anonymous class instead of using `Tag.of(splitTag[0], splitTag[1])`?

Reply:
You're right, much simpler that way. Thanks :)